packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_loan_set_loan_token.test.ts

```diff
@@ -0,0 +1,325 @@
+  })
+
+  it('should not setLoanToken if token with same symbol was created before', async () => {
+    const priceFeedId = await testing.container.call('appointoracle', [await testing.generateAddress(), [{
+      token: 'Token3',
+      currency: 'USD'
+    }], 1])
+    await testing.generate(1)
+
+    await testing.rpc.loan.setLoanToken({
+      symbol: 'Token3',
+      priceFeedId
+    })
+    await testing.generate(1)
+
+    // Fund 10 DFI UTXO
+    await fundEllipticPair(testing.container, providers.ellipticPair, 10)
+    await providers.setupMocks() // Required to move utxos
+
+    const script = await providers.elliptic.script()
+    const txn = await builder.loans.setLoanToken({
+      symbol: 'Token3',
+      name: 'Token3',
+      priceFeedId,
+      mintable: true,
+      interest: new BigNumber(3.5)
+    }, script)
+
+    const promise = sendTransaction(testing.container, txn)
+    await expect(promise).rejects.toThrow('DeFiDRpcError: \'LoanSetLoanTokenTx: token \'Token3\' already exists! (code 16)\', code: -26')
+  })
+
+  it('should not setLoanToken if priceFeedId does not contain USD price', async () => {
+    const priceFeedId: string = await testing.container.call('appointoracle', [await testing.generateAddress(), [{
+      token: 'Token4',
+      currency: 'Token4'
+    }], 1])
+    await testing.generate(1)
+
+    const script = await providers.elliptic.script()
+    const txn = await builder.loans.setLoanToken({
+      symbol: 'Token4',
+      name: 'Token4',
+      priceFeedId,
+      mintable: true,
+      interest: new BigNumber(4.5)
+    }, script)
+    const promise = sendTransaction(testing.container, txn)
+
+    await expect(promise).rejects.toThrow(`DeFiDRpcError: 'LoanSetLoanTokenTx: oracle (${priceFeedId}) does not contain USD price for this token! (code 16)', code: -26`)
+  })
+
+  it('should not setLoanToken if priceFeedId is invalid', async () => {
+    const script = await providers.elliptic.script()
+    const txn = await builder.loans.setLoanToken({
+      symbol: 'Token5',
+      name: 'Token5',
+      priceFeedId: 'e40775f8bb396cd3d94429843453e66e68b1c7625d99b0b4c505ab004506697b',
+      mintable: true,
+      interest: new BigNumber(5.5)
+    }, script)
+
+    const promise = sendTransaction(testing.container, txn)
+    await expect(promise).rejects.toThrow('DeFiDRpcError: \'LoanSetLoanTokenTx: oracle (e40775f8bb396cd3d94429843453e66e68b1c7625d99b0b4c505ab004506697b) does not exist or not valid oracle! (code 16)\', code: -26')
+  })
+
+  it('should setLoanToken with the given name', async () => {
+    const priceFeedId = await testing.container.call('appointoracle', [await testing.generateAddress(), [{
+      token: 'Token6',
+      currency: 'USD'
+    }], 1])
+    await testing.generate(1)
+
+    const script = await providers.elliptic.script()
+    const txn = await builder.loans.setLoanToken({
+      symbol: 'Token6',
+      name: 'Token6',
```

Comment:
`name = ''`
can add this to a new test or embed to an existing test

Reply:
Changed.